server/index.ts

```diff
@@ -194,6 +227,15 @@ router.post('/message', async (req, res) => {
     version: '2'
   });
 
+  if (msg.type === 'delete-proposal') {
+    let query = `SELECT address FROM messages WHERE type = 'proposal' AND id = ?`;
+    let propasalSigner = await db.queryAsync(query, [msg.payload.proposal]);
+    if (propasalSigner[0].address !== body.address) {
+      return sendError(res, 'wrong signer');
+    }
+    await archiveProposal(msg.payload.proposal);
```

Comment:
Actually i mistaken in my previous comment, you was right, the `SELECT` query to check if the action is valid should happen before the `pinJson` function and the `archiveProposal` should happen after, so we need `if (msg.type === 'delete-proposal') {` in 2 different place. Sorry for the hassle!